jina/executors/__init__.py

```diff
@@ -147,6 +157,7 @@ def _post_init_wrapper(self, _metas: Dict = None, _requests: Dict = None, fill_i
 
             _before = set(list(vars(self).keys()))
             self.post_init()
+            self._check_on_gpu()
```

Comment:
I recommend in the following order as some user may call gpu library in `post_init` already

```suggestion
            self._check_on_gpu()
            self.post_init()
```